h/static/scripts/annotator/plugin/pdf-metadata.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/**
+ * This PDFMetadata service provides implementations for extracting fields
+ * of a pdf document from the input PDFViewerApplication object.
+ *
+ * The fields extracted are:
+ * #1. urn - a string which uniquely identifies a pdf document.
+ * #2. title - a string representing the title of the pdf document.
+ * #3. link - an array of two objects, the urn and the url strings of the pdf.
+ * #4. documentFingerprint - the string representing the pdf fingerprint.
+ */
+function PDFMetadata(PDFViewerApplication) {
```

Comment:
I'd suggest that you don't overload the global `PDFViewerApplication` with the name of the parameter. Call it `application` or even `app`.